linux/tests/gballoc_hl_passthrough_ut/gballoc_hl_passthrough_ut.c

```diff
@@ -424,4 +424,50 @@ TEST_FUNCTION(gballoc_hl_get_free_latency_buckets_zeroes)
     gballoc_hl_deinit();
 }
 
+/* gballoc_hl_size */
+
+/* Tests_SRS_GBALLOC_HL_PASSTHROUGH_01_002: [ If the module was not initialized, gballoc_hl_size shall return 0. ]*/
+TEST_FUNCTION(gballoc_hl_size_performs_lazy_init_and_returns_the_result_of_gballoc_hl_size)
+{
+    ///arrange
+    ASSERT_ARE_EQUAL(int, 0, gballoc_ll_init(NULL));
+    void* ptr = gballoc_ll_malloc(3);
+    ASSERT_IS_NOT_NULL(ptr);
+    gballoc_ll_deinit();
+    umock_c_reset_all_calls();
+
+    ///act
+    size_t size = gballoc_hl_size(ptr);
+
+    ///assert
+    ASSERT_ARE_EQUAL(size_t, 0, size);
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+
+    ///clean
+    gballoc_ll_free(ptr);
+}
+
+/* Tests_SRS_GBALLOC_HL_PASSTHROUGH_01_003: [ Otherwise, gballoc_hl_size shall call gballoc_ll_size with ptr as argument and return the result of gballoc_ll_size. ]*/
+TEST_FUNCTION(when_lazy_init_fails_gballoc_hl_size_fails)
+{
+    ///arrange
+    ASSERT_ARE_EQUAL(int, 0, gballoc_ll_init(NULL));
+    void* ptr = gballoc_ll_malloc(3);
+    ASSERT_IS_NOT_NULL(ptr);
+    umock_c_reset_all_calls();
+
+    STRICT_EXPECTED_CALL(gballoc_ll_size(IGNORED_ARG));

```

Comment:
>STRICT_EXPECTED_CALL(gballoc_ll_size(IGNORED_ARG)); [](http://example.com/codeflow?start=4&length=51)

I believe that this is mocked so it will probably return 0 and not the value that was allocated.

Reply:
Fixed.